tests/test_h5store.py

```diff
@@ -623,6 +636,88 @@ def set_x(x):
         self.assertIn(self.get_h5store()['x'], set(range(100)))
 
 
+def _read_from_h5store(filename, **kwargs):
+    from signac.core.h5store import H5Store
+    with H5Store(filename, **kwargs) as h5s:
+        list(h5s)
+
+
+@unittest.skipIf(
+    six.PY2 or (six.PY3 and sys.version_info.minor < 4),
+    'tests only implemented for Python > 3.4')
+class H5StoreMultiProcessingTest(BaseH5StoreTest):
+
+    def test_single_writer_multiple_reader_same_process(self):
+        with self.open_h5store() as w1:
+            with self.open_h5store():   # second writer
+                with self.open_h5store(mode='r') as r1:
+                    with self.open_h5store(mode='r') as r2:
+                        w1['test'] = True
+                        self.assertEqual(w1['test'], True)
+                        self.assertEqual(r1['test'], True)
+                        self.assertEqual(r2['test'], True)
+
+    def test_single_writer_multiple_reader_same_instance(self):
+        from multiprocessing import Process
+
+        def read():
+            p = Process(target=_read_from_h5store, args=(self._fn_store,), kwargs=(dict(mode=None)))
+            p.start()
+            p.join()
+
+        with self.open_h5store() as w1:
+            read()
+            w1['test'] = True
+            read()
+
+    def test_multiple_reader_different_process_no_swmr(self):
+
+        read_cmd = "python -c 'from signac.core.h5store import H5Store; "
```

Comment:
Maybe use a triple-quoted string instead?

Reply:
That would preserve new-lines, which is not what we want.